### PR TITLE
chore: revamp the frontend architecture 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Refactor 

This PR takes care of moving away from redux as much as possible and streamlining the routing , sync setup calls and making sure we have deterministic actions based on API results. All the test cases have been fixed in sync with the latest design architecture 

Contributes to - https://github.com/SigNoz/signoz/issues/5621 , https://github.com/SigNoz/engineering-pod/issues/2068

### CODE REVIEW

**Comment by vikrantgupta25** [comment]
contributes to - https://github.com/SigNoz/engineering-pod/issues/2068

**Comment by vikrantgupta25** [comment]
contributes to - https://github.com/SigNoz/engineering-pod/issues/1957

**Comment by srikanthccv** [comment]
https://app.intercom.com/a/inbox/wmkurvb7/inbox/admin/6745296/conversation/5813#part_id=comment-5813-17320 more and more users are facing this and perhaps it became more visible and frequent

**Comment by srikanthccv** [comment]
https://github.com/SigNoz/engineering-pod/issues/2096

**Comment by vikrantgupta25** [comment]
contributes to - https://github.com/SigNoz/signoz/issues/5621

File: frontend/src/AppRoutes/index.tsx [hunk 1]
```diff
@@ -51,179 +35,135 @@ import defaultRoutes, {
...
+				setLocalStorageApi(LOCALSTORAGE.IS_IDENTIFIED_USER, 'true');
+			}
 
-			setRoutes(newRoutes);
+			let updatedRoutes = defaultRoutes;
+			// if the user is a cloud user
+			if (isCloudUserVal || isEECloudUser()) {
+				// if the user is on basic plan then remove billing
+				if (isOnBasicPlan) {
+					updatedRoutes = updatedRoutes.filter(
```
Comment (YounixM) [inline frontend/src/AppRoutes/index.tsx:144]: @vikrantgupta25: Billing should be should only to admins, do we have a check somewhere else ?

File: frontend/src/container/AppLayout/index.tsx [hunk 2]
```diff
@@ -248,15 +234,16 @@ function AppLayout(props: AppLayoutProps): JSX.Element {
...
-			licenseData?.payload?.onTrial &&
-			!licenseData?.payload?.trialConvertedToSubscription &&
-			!licenseData?.payload?.workSpaceBlock &&
-			getRemainingDays(licenseData?.payload.trialEnd) < 7
+			!isFetchingLicenses &&
+			licenses &&
+			licenses.onTrial &&
+			!licenses.trialConvertedToSubscription &&
+			!licenses.workSpaceBlock &&
+			getRemainingDays(licenses.trialEnd) < 7
```
Comment (YounixM) [inline frontend/src/container/AppLayout/index.tsx:242]: NIT: destruct licenses and move 7 to const

File: frontend/src/container/AppLayout/index.tsx [hunk 3]
```diff
@@ -272,11 +259,12 @@ function AppLayout(props: AppLayoutProps): JSX.Element {
 		// after logging out hide the trial expiry banner
 		if (!isLoggedIn) {
 			setShowTrialExpiryBanner(false);
+			setShowPaymentFailedWarning(false);
 		}
 	}, [isLoggedIn]);
 
 	const handleUpgrade = (): void => {
-		if (role === 'ADMIN') {
+		if (user.role === 'ADMIN') {
```
Comment (YounixM) [inline frontend/src/container/AppLayout/index.tsx:267]: safety check

File: frontend/src/container/IngestionSettings/IngestionSettings.tsx [hunk 4]
```diff
@@ -3,18 +3,16 @@ import './IngestionSettings.styles.scss';
...
-import AppReducer from 'types/reducer/app';
 
 export default function IngestionSettings(): JSX.Element {
-	const { user } = useSelector<AppState, AppReducer>((state) => state.app);
+	const { user } = useAppContext();
 
 	const { data: ingestionData, isFetching } = useQuery({
 		queryFn: getIngestionData,
-		queryKey: ['getIngestionData', user?.userId],
+		queryKey: ['getIngestionData', user?.id],
```
Comment (YounixM) [inline frontend/src/container/IngestionSettings/IngestionSettings.tsx:15]: add enabled check here.

File: frontend/src/container/ListAlertRules/ListAlert.tsx [hunk 5]
```diff
@@ -42,12 +40,11 @@ const { Search } = Input;
 
 function ListAlert({ allAlertRules, refetch }: ListAlertProps): JSX.Element {
 	const { t } = useTranslation('common');
-	const { role, featureResponse } = useSelector<AppState, AppReducer>(
-		(state) => state.app,
-	);
+	const { user } = useAppContext();
+	// TODO[vikrantgupta25]: check with sagar on cleanup
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/ListAlert.tsx:44]: cc : @SagarRajput-7 @vikrantgupta25

File: frontend/src/container/ListOfDashboard/ImportJSON/index.tsx [hunk 6]
```diff
@@ -239,12 +225,6 @@ function ImportJSON({
 						>
 							{t('import_and_next')} &nbsp; <MoveRight size={14} />
 						</Button>
-
-						{isFeatureAlert && (
```
Comment (YounixM) [inline frontend/src/container/ListOfDashboard/ImportJSON/index.tsx:243]: Was this not being consumed ?

**Review by YounixM — DISMISSED** [review]
Minor Comments, can be taken up in a separate PR.

File: frontend/src/AppRoutes/index.tsx (hunk 1 above)
Comment (vikrantgupta25) [reply frontend/src/AppRoutes/index.tsx:144]: yes we have the same in `SideNav.tsx`

File: frontend/src/container/AppLayout/index.tsx (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/container/AppLayout/index.tsx:267]: not required as the user.role will always be present [ default is VIEWER ]

File: frontend/src/container/ListOfDashboard/ImportJSON/index.tsx (hunk 6 above)
Comment (vikrantgupta25) [reply frontend/src/container/ListOfDashboard/ImportJSON/index.tsx:243]: no we removed usage based features so removed the check for everywhere

File: frontend/src/container/ListAlertRules/ListAlert.tsx (hunk 5 above)
Comment (vikrantgupta25) [reply frontend/src/container/ListAlertRules/ListAlert.tsx:44]: not required anymore, i cleaned it up ✅

Comment (vikrantgupta25) [reply frontend/src/container/ListAlertRules/ListAlert.tsx:44]: will remove the TODO in follow up

File: frontend/src/container/IngestionSettings/IngestionSettings.tsx (hunk 4 above)
Comment (vikrantgupta25) [reply frontend/src/container/IngestionSettings/IngestionSettings.tsx:15]: will check in follow up PR , didn't update the existing code logics

**Review by ahmadshaheer — APPROVED** [review]

File: frontend/src/tests/test-utils.tsx [hunk 7]
```diff
@@ -77,35 +88,277 @@
...
+				updated_at: '0',
+			},
+			key: 'does-not-matter',
+			state: LicenseState.ACTIVE,
+			status: LicenseStatus.VALID,
+		},
+		isFetchingActiveLicenseV3: false,
+		activeLicenseV3FetchError: null,
+		user: {
+			accessJwt: 'some-token',
```
[the remark this answers is not in the document]
Comment (vikrantgupta25) [reply frontend/src/tests/test-utils.tsx:111]: Dummy token!

**Review by ahmadshaheer — APPROVED** [review]

**Review by srikanthccv — APPROVED** [review]
I am approving to unblock.

**Review by SagarRajput-7 — APPROVED** [review]

**Pull request merged**